src/main/java/com/microsoft/sqlserver/jdbc/DDC.java

```diff
@@ -1060,6 +1064,136 @@ static final Object convertTemporalToObject(JDBCType jdbcType, SSType ssType, Ca
         }
     }
 
+    private static Object convertTemporalToObject(JDBCType jdbcType, SSType ssType, int daysSinceBaseDate,
+            long ticksSinceMidnight, int fractionalSecondsScale) {
+        int subSecondNanos;
+
+        // In cases where Timezone values don't need to be considered, use LocalDateTime go avoid
+        // overhead from using a Calendar object.
+        // Note that DateTimeOffset path is not handled in this method, since DateTimeOffset always has
+        // its own Calendar object (UTC timezone) associated with it.
+        LocalDateTime ldt = null;
+
+        switch (ssType) {
+            case TIME: {
+                ldt = LocalDateTime.of(TDS.BASE_YEAR_1900, 1, 1, 0, 0, 0).plusNanos(ticksSinceMidnight);
+
+                subSecondNanos = (int) (ticksSinceMidnight % Nanos.PER_SECOND);
+                break;
+            }
+
+            case DATE:
+            case DATETIME2:
+            case DATETIMEOFFSET: {
+                ldt = LocalDateTime.of(1, 1, 1, 0, 0, 0);
+                ldt = ldt.plusDays(daysSinceBaseDate);
+                if (jdbcType.category != JDBCType.Category.DATE && ssType != SSType.DATE) {
+                    ldt = ldt.plusNanos(ticksSinceMidnight);
+                }
+                subSecondNanos = (int) (ticksSinceMidnight % Nanos.PER_SECOND);
+                break;
+            }
+
+            case DATETIME: // and SMALLDATETIME
+            {
```

Comment:
why don't we need to add "case SMALLDATETIME:" here?

Reply:
This comment is misleading, but basically the cases are based off of SSType.Category. DATETIME and SMALLDATETIME both go under Category.DATETIME, so we don't need to add a case for SMALLDATETIME.